classes/Combination.php

```diff
@@ -424,7 +425,9 @@ public static function getIdByReference($idProduct, $reference)
         $query->select('pa.id_product_attribute');
         $query->from('product_attribute', 'pa');
         $query->where('pa.reference LIKE \'%' . pSQL($reference) . '%\'');
-        $query->where('pa.id_product = ' . (int) $idProduct);
+        if (!empty($idProduct)) {
+            $query->where('pa.id_product = ' . (int) $idProduct);
```

Comment:
The reference is not unique can returns an array.
The reference use like, it's not a good idea.
What you probably want is to add a new function
```php
    /**
     * For a given product_attribute reference, returns the product ids.
     *
     * @param string $reference
     *
     * @return int id
     */
    public static function getByReference($reference)
    {
        if (empty($reference)) {
            return [];
        }

        $query = new DbQuery();
        $query->select('pa.id_product_attribute, pa.id_product');
        $query->from('product_attribute', 'pa');
        $query->where('pa.reference = \'' . pSQL($reference) . '\'');

        return Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($query);
    }
```

Reply:
Hello

I can rework this code but these last remarks don't look related to this PR. Theses notes concern yet the actual code ( https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Combination.php#L442-L455) . 
Like is yet used on query and we can have many reference with this query as any unique index are set on id_product/reference.

Do you confirm the need of this code rework ?

Reply:
Yes, I confirm it needs to be refactor what you want is something a little different than the original function (which is badly named).

Reply:
Hello

Understood. I rework on this.